gpjax/dataset.py

```diff
@@ -84,7 +83,7 @@ def out_dim(self) -> int:
         return self.y.shape[1]
 
 
-def _check_shape(X: Float[Array, "N D"], y: Float[Array, "N Q"]) -> None:
+def _check_shape(X: Optional[Float[Array, "..."]], y: Optional[Float[Array, "..."]]) -> None:
```

Comment:
Here I removed the beartype-shape checking. Could instead simply remove the `_check_shape` function...

Reply:
Is there any reason not to remove `_check_shape` now we use beartype? If not, then let's do it.

Reply:
Actually, beartype only checks arguments individually, and does not check consistency of dimensions across multiple arguments/return values, so might be better to keep the `_check_shape` (so e.g. "N1 D" and "N2 D" shapes get flagged as not-matching).

Reply:
Actually actually, jaxtyping itself checks those, I just had a bug in my toy example trying it out. But the error messages are still more verbose/less precise than the ones emitted by _check_shape :disappointed:

Reply:
OK. I'm guessing there's no easy way for us to customise the error messages that Beartype throws?

Reply:
If the answer to my above question is no, then let's keep `_check_shape`.

Reply:
It looks like this is a *very* hot topic right now: https://github.com/google/jaxtyping/issues/6.